### PR TITLE
SMP Release

### PR DESCRIPTION
**Pull request by n1xan**

### CODE REVIEW

File: bellatrix.web/src/main/java/solutions/bellatrix/web/components/contracts/ComponentBackgroundColor.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+public interface ComponentBackgroundColor extends Component {
+
+    default String getBackgroundColor(){
+        var script = "return arguments[0].style.background";
+        var result = new JavaScriptService().execute(script, (WebComponent) this);
+
+        return result;
+    }
+
+    @SneakyThrows
```
Comment (n1xan) [inline bellatrix.web/src/main/java/solutions/bellatrix/web/components/contracts/ComponentBackgroundColor.java:36]: delete sneaky throws

File: bellatrix.web/src/main/java/solutions/bellatrix/web/components/contracts/ComponentStyle.java [hunk 2]
```diff
@@ -49,4 +51,21 @@ default void validateStyleContains(String value) {
...
     }
+
+    default String getStyle(CssStyle style) {
+        var script = String.format("return window.getComputedStyle(arguments[0],null).getPropertyValue('%s');", style);
+        var result = new JavaScriptService().execute(script, (WebComponent) this);
+
+        return result;
+    }
+
+    @SneakyThrows
```
Comment (n1xan) [inline bellatrix.web/src/main/java/solutions/bellatrix/web/components/contracts/ComponentStyle.java:62]: delete sneaky throws

File: bellatrix.core/src/main/java/solutions/bellatrix/core/assertions/EntitiesAsserter.java [hunk 3]
```diff
@@ -61,25 +61,32 @@ private static <TEntity> List<Exception> assertAreEqualsInternal(TEntity expecte
                 try {
                     currentExpectedProperty = expectedObject.getClass().getMethod(currentRealProperty.getName());
                 } catch (NoSuchMethodException e) {
-                    failedAssertions.add(e);
+                    System.out.println(String.format("Property %s not found.", currentRealProperty));
+//                    failedAssertions.add(e);
```
Comment (n1xan) [inline bellatrix.core/src/main/java/solutions/bellatrix/core/assertions/EntitiesAsserter.java:65]: remove line

File: bellatrix.web/src/main/java/solutions/bellatrix/web/components/WebComponent.java [hunk 4]
```diff
@@ -1033,19 +1046,20 @@ private void addArtificialDelay() {
...
         // createBy default scroll down to make the element visible.
         if (webSettings.getAutomaticallyScrollToVisible()) {
-            scrollToVisible(wrappedElement, false);
+            scrollToVisible(wrappedElement, false, "center");
         }
     }
 
-    private void scrollToVisible(WebElement wrappedElement, boolean shouldWait) {
+
+    private void scrollToVisible(WebElement wrappedElement, boolean shouldWait, String scrollPosition) {
```
Comment (MiriamKyoseva) [inline bellatrix.web/src/main/java/solutions/bellatrix/web/components/WebComponent.java:1054]: extract into enum scrollPosition

File: bellatrix.core/src/main/java/solutions/bellatrix/core/utilities/SingletonFactory.java [hunk 5]
```diff
@@ -37,6 +37,7 @@ public static <T> T getInstance(Class<T> classOf, Object... initargs) {
             return (T)SINGLETON_FACTORY.mapHolder.get(classOf.getName());
         } catch (Exception e) {
             // not the best practice to return null. But probably we will never end here so it is OK.
+            Log.error("Failed to create instance of the page. Exception was: " + e);
```
Comment (MiriamKyoseva) [inline bellatrix.core/src/main/java/solutions/bellatrix/core/utilities/SingletonFactory.java:40]: Shouldn't the message be more generic? Maybe get the name of the class we tried to get an instance of and use String.format

File: bellatrix.web/src/main/java/solutions/bellatrix/web/components/WebComponent.java [hunk 6]
```diff
@@ -148,6 +157,10 @@ public String getTitle() {
         return getAttribute("title");
     }
 
+    public WebComponent getParent() {
```
Comment (MiriamKyoseva) [inline bellatrix.web/src/main/java/solutions/bellatrix/web/components/WebComponent.java:160]: Sounds ambiguous with parentWrappedElement field and its get method, immediate difference being only the type

File: bellatrix.web/src/main/java/solutions/bellatrix/web/components/contracts/ComponentChecked.java [hunk 7]
```diff
@@ -30,6 +30,16 @@ default void validateIsChecked() {
         ComponentValidator.defaultValidateAttributeTrue((WebComponent)this, this::isChecked, "checked");
     }
 
+    @SneakyThrows
```
Comment (MiriamKyoseva) [inline bellatrix.web/src/main/java/solutions/bellatrix/web/components/contracts/ComponentChecked.java:33]: SneakyThrows not needed

File: bellatrix.web/src/main/java/solutions/bellatrix/web/pages/PageMap.java [hunk 8]
```diff
@@ -13,10 +13,15 @@
...
 import solutions.bellatrix.web.services.ComponentCreateService;
 
 public abstract class PageMap {
     public ComponentCreateService create() {
-        return new ComponentCreateService();
+        return app().create();
+    }
+
+    public App app() {
+        return new App();
```
Comment (MiriamKyoseva) [inline bellatrix.web/src/main/java/solutions/bellatrix/web/pages/PageMap.java:25]: Shouldn't this be SingletonFactory.getInstance(App.class) ?

**Review by MiriamKyoseva — APPROVED** [review]

**Pull request merged**